wnfs/src/private/directory.rs

```diff
@@ -1251,15 +1302,66 @@ impl PrivateDirectory {
         }
 
         Ok(Self {
+            persisted_as: OnceCell::new_with(Some(from_cid)),
             version,
             metadata,
             header: {
                 let cbor_bytes = key.0.decrypt(&header).map_err(DeError::custom)?;
                 dagcbor::decode(&cbor_bytes).map_err(DeError::custom)?
             },
+            previous,
             entries,
         })
     }
+
+    pub async fn store<B: BlockStore>(&self, store: &mut B, rng: &mut impl RngCore) -> Result<Cid> {
```

Comment:
Nit: Thinking we should stick to a convention here.

```rs
pub async fn store<B: BlockStore, R: RngCore>(&self, store: &mut B, rng: &mut R) -> Result<Cid>;
```

```rs
pub async fn store<B, R>(&self, store: &mut B, rng: &mut R) -> Result<Cid> where B: BlockStore, R: RngCore;
```

Reply:
Generally I think the fewer type variables the better.
What do you think of

```rs
pub async fn store(&self, store: &mut impl BlockStore, rng: &mut impl RngCore) -> Result<Cid> {
```

?